### DISCUSSION TITLE
Hide window decorations / titlebar and set window icon

### DISCUSSION THREAD

**Discussion by siriuslee69** [webui-dev/webui#150] [Ideas]
I would love to see an option to hide the entire titlebar and or choose your own window icon as well as window name. :) Would that be possible?

**Comment by ttytm** [comment]
Hey @siriuslee69 👋 

I'm afraid there is no gernral approach. At least noone I know about. What OS are you using?
I'm asking because often it's the window manager and compositor that is running that is responsible for such things not an application. And it depends a bit if and how this is possible.

**Reply by siriuslee69** [reply]
Hey, thanks for the quick reply! I'm using Windows 11 and was thinking about using Vivaldi for the window creation! 
I was thinking, maybe I could somehow get the window ID from the Win32 API and then change the decorations after window creation.
But ChatGPT told me this wouldnt be possible, since usually apps can't access other apps' windows and change them.  

Maybe I could host the browser window inside another window without a frame and then enter borderless fullscreen mode with F11? Or do you maybe have any workarounds to achieve something like that?

**Reply by ttytm** [reply]
Thanks for the Info.
I unfortunately can't offer good help here as I'm no expert on Windows, I gave it up years ago.
I can only guess that it will require to hack your way around it somehow. As you already state that it's the browser window that is used it's probably necessary to embed it inside another window. If full screen doesn't work maybe adding a negative top positioning to hide the title bar will work. Maybe there is a browser that supports settings for hiding the titlebar, vivaldi is very comprehensive in it's settings afaik.

I remember from other repos that this behavior wasn't supported on Windows (e.g. https://github.com/neovide/neovide/issues/1756). So the best thing I can personally recommend is to not use windows - if that's an option - that will generally save a lot of headache (although, not gonna lie, a change will also cause some headache at first 😅). On linux it's my default setting not to use title bars. It's usually possible to disable them directly via system settings or there are simple extensions for it.

**Comment by JOTSR** [comment]
> choose your own window icon as well as window name.
I'm not understanding your probleme. By setting `<title>` and `<link rel="icon" ...>` you can customize your window title and icon.
![image](https://github.com/webui-dev/webui/assets/58234376/b3c847c6-04ad-44af-a1ce-e95569b3ea58)
Source example [Deplot](https://deno.land/x/deplot@1.1.0).
Note that the process name is that of the used browser.

> I would love to see an option to hide the entire titlebar
It is a good suggestion and I think of two approaches:
- Deals with os API (it seams complex to use and port)
- Use [window control API provided to PWA](https://web.dev/window-controls-overlay/) and provides end user more customization over the window (need to find a way to auto install and remove pwa on ui launch) + it fix the process name issue

**Reply by siriuslee69** [reply]
Wow, thank you so much lol! I'm so used to the website title and icon displaying inside the tab instead of the window that I didn't realize you could change them like this.
I will still have to read up on the PWA stuff though. 
Thanks again!

**Reply by ttytm** [reply]
Oh that's a nice approach @JOTSR. It also makes it into another great advantage that a browser is used.

Sorry, I missed on the second part of the question as well. Somehow I was stuck with unsuccessfully hiding window titlbars on windows 😅.

**Reply by AlbertShown** [reply]
Removing the titlbar will be available soon, but it only works in Chromium-based browsers.

More info:
https://github.com/webui-dev/webui/issues/128#issuecomment-1634895648
